nautobot/core/views/renderers.py

```diff
@@ -99,7 +99,7 @@ def construct_table(self, view, **kwargs):
             return RequestConfig(request, paginate).configure(table)
         else:
             pk_list = kwargs.get("pk_list", [])
-            table = table_class(queryset.filter(pk__in=pk_list), orderable=False)
+            table = table_class(view.get_queryset().filter(pk__in=pk_list), orderable=False)
```

Comment:
Why not `alter_queryset` here too?

Reply:
The else statement here would apply to the `bulk_destroy` and `bulk_update` actions. In the legacy views `alter_queryset` is only called on the `ObjectListView` view and it's not being called on the detail view notes or changelog tabs. However, `alter_queryset` is defined on `NautobotViewSetMixin` so it is available for all of the actions (`retrieve`, `list`, `destroy`, `create`, `update`, `bulk_destroy`, `bulk_create`, `bulk_update`, `changelog` and `notes`). I'm leaning toward calling it for every action here?

Also, the only documentation I'm seeing for `alter_queryset` is on [this line](https://github.com/nautobot/nautobot/blob/5db9249dad4a18cc66ca6794dfbf8c797c1e2cef/nautobot/core/views/generic.py#L254). We should probably document the `NautobotViewSetMixin.alter_queryset()` method with its intended use cases.